scripts/globals/porter_moogle_util.lua

```diff
@@ -0,0 +1,311 @@
+                storableItemIds[#storableItemIds+1] = slotItemId;
+            end
+        end
+    end
+    
+    return storableItemIds;
+end
+
+----------------------------------------------------------------------
+-- desc : Stores the items on the Storage Slip extra data and starts
+--        the event indicating that the storage was successful.
+----------------------------------------------------------------------
+function storeItems(player, storableItemIds, slipId, e)
+    if (#storableItemIds > 0) then
+        local param0 = 0x0;
+        local param1 = 0x0;
+        if (#storableItemIds == 1) then
+            param0 = storableItemIds[1];
+            param1 = 0x00;
+        else
+            param0 = #storableItemIds;
+            param1 = 0x01;
+        end
+        
+        -- idk
+        local extra = { };
+        for i = 0, 23 do
+            extra[i] = 0;
+        end
+        
+        for k, v in ipairs(slipItems[slipId]) do
+            if find(storableItemIds, v) ~= nil then
+                local bitmask = extra[math.floor((k - 1) / 8)];
+                if bitmask < 0 then
+                    bitmask = bitmask + 256;
+                end
+                extra[math.floor((k - 1) / 8)] = bit.bor(bitmask, bit.lshift(1, (k - 1) % 8));
+            end
+        end
+        
+        local result = player:storeWithPorterMoogle(slipId, extra, storableItemIds);
+        
+        if (result == 0) then
+            player:startEvent(e.STORE_EVENT_ID, param0, param1);
+        elseif (result == 1) then
+            player:startEvent(e.ALREADY_STORED_ID);
+        elseif (result == 2) then
+            player:startEvent(e.MAGIAN_TRIAL_ID);
+        end
+    end
+end
+
+----------------------------------------------------------------------
+-- desc : Returns a zero-based identifier for the slip (Storage Slip 1
+--        is index 0, Storage Slip 2 is index 1, etc).
+----------------------------------------------------------------------
+function getSlipIndex(slipId)
+    return find(slipIds, slipId) - 1;
+end
+
+----------------------------------------------------------------------
+-- desc : Gets the extra data from the traded slip and starts the 
+--        retrieval event.
+----------------------------------------------------------------------
+function startRetrieveProcess(player, eventId, slipId)
+    local extra = player:getRetrievableItemsForSlip(slipId);
+    local params = int8ToInt32(extra);
+    local slipIndex = getSlipIndex(slipId);
+    
+    player:setLocalVar('slipId', slipId);
+    player:startEvent(eventId, params[1], params[2], params[3], params[4], params[5], params[6], nil, slipIndex);   
+end
+
+----------------------------------------------------------------------
+-- desc : Begins the storage or retrieval process based on the items
+--        supplied in the trade.
+----------------------------------------------------------------------
+function doTrade(player, trade, e)
```

Comment:
since these are global, it's probably best to give them a name related to porter moogles directly (porterDoTrade, etc)